packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveCmp.ts

```diff
@@ -39,20 +39,33 @@ import {
 export class ForceSourceRetrieveExecutor extends SfdxCommandletExecutor<
   string
 > {
-  private metadataArg: string;
-  constructor(metadataArg: string) {
+  private typeName: string;
+  private componentName: string;
+
+  constructor(typeName: string, componentName: string) {
     super();
-    this.metadataArg = metadataArg;
+    this.typeName = typeName;
+    this.componentName = componentName;
   }
 
   public build(): Command {
     return new SfdxCommandBuilder()
       .withDescription(nls.localize('force_source_retrieve_text'))
       .withArg('force:source:retrieve')
-      .withFlag('-m', this.metadataArg)
+      .withFlag('-m', `${this.typeName}:${this.componentName}`)
```

Comment:
I changed this so I could access just the type name and not the whole string we use for the command